src/expr/proof_node_updater.cpp

```diff
@@ -125,28 +125,21 @@ void ProofNodeUpdater::processInternal(
       }
       traversing.push_back(cur);
       visit.push_back(cur);
-      if (d_mergeSubproofs)
+      // If we are not the top-level proof, we were a scope, or became a scope
```

Comment:
We should still check d_mergeSubproofs here, right?

Reply:
In that case we'd do the tracking only when we are merging subproofs. The way I see it this tracking serves for more than doing the merges correctly, so I didn't want to have it constrained by this.

Reply:
Ah, I see. 
Yes, I think this was a micro-optimization that we don't need, I agree with the change.